### ISSUE TITLE
i don't understand code: torch.mean(F.softplus(logz_unlabel) why add F.softplus activation?

### ISSUE THREAD

**Issue by daixiangzi** [Sleepychord/ImprovedGAN-pytorch#4]

**Comment by fishishappyandfree** [comment]
do you run the code? could you tell me the accuracy on the test data by using the default configs?

**Comment by fishishappyandfree** [comment]
your problem is mentioned in the README.md

**Comment by daixiangzi** [comment]
> do you run the code? could you tell me the accuracy on the test data by using the default configs?

sorry, i don't run the code,i only read this code,

**Comment by Sleepychord** [comment]
Hi, @daixiangzi this is a technique from improved gan. As showed in this latex picture, 

<img width="956" alt="image" src="https://user-images.githubusercontent.com/9153807/65815396-6783cd80-e221-11e9-9f7a-13fb34de1521.png">

The 0.5 results from the aim to balance the size of supervised and unsupervised data. We have a batch of unlabeled real data and a batch of fake data for unsupervised learning, but only a batch of labeled data for supervised learning.

**Comment by Sleepychord** [comment]
The softplus replacement mentioned in the README.md by @fishishappyandfree is about the codes in `forward` of `Generator`. It is just to replace the activation function and has no relevance to the codes which you mentioned.

**Comment by daixiangzi** [comment]
> The softplus replacement mentioned in the README.md by @fishishappyandfree is about the codes in `forward` of `Generator`. It is just replace a activation function and has no relevance to the code which you mentioned.

ok,thanky you ,i know
